Package.swift

```diff
@@ -18,24 +18,33 @@ func resolveDependencies() -> [Package.Dependency] {
 }
 
 func resolveTargets() -> [Target] {
+    let objcSources = ["Purchases/Info.plist",
+                       "Purchases/Caching",
+                       "Purchases/FoundationExtensions",
+                       "Purchases/Misc",
+                       "Purchases/Networking",
+                       "Purchases/Public",
+                       "Purchases/Purchasing",
+                       "Purchases/ProtectedExtensions",
+                       "Purchases/SubscriberAttributes"]
+    let infoPlist = "Purchases/Info.plist"
+    let swiftSources = "Purchases/SwiftSources"
+    
     let baseTargets: [Target] = [
         .target(name: "Purchases",
-                dependencies: [],
+                dependencies: ["PurchasesSwift"],
                 path: ".",
-                exclude: ["Purchases/Info.plist"],
+                exclude: [infoPlist, swiftSources],
                 sources: ["Purchases"],
                 publicHeadersPath: "Purchases/Public",
-                cSettings: [
-                    .headerSearchPath("Purchases"),
-                    .headerSearchPath("Purchases/Caching"),
-                    .headerSearchPath("Purchases/FoundationExtensions"),
-                    .headerSearchPath("Purchases/Misc"),
-                    .headerSearchPath("Purchases/Networking"),
-                    .headerSearchPath("Purchases/Public"),
-                    .headerSearchPath("Purchases/Purchasing"),
-                    .headerSearchPath("Purchases/ProtectedExtensions"),
-                    .headerSearchPath("Purchases/SubscriberAttributes"),
-    ])]
+                cSettings: objcSources.map { CSetting.headerSearchPath($0) }
+        ),
+        .target(name: "PurchasesSwift",
+                dependencies: [],
+                path: ".",
+                exclude: [infoPlist] + objcSources,
+                sources: ["Purchases"],
+                publicHeadersPath: swiftSources)]
```

Comment:
we define two targets, one with only Obj-C files, one with only Swift files, but only one product, so to end users it's the same as before.

Reply:
this does mean that swift files won't ever be public through SPM, though.